### ISSUE TITLE
Error in pix2pix backward_G?

### ISSUE THREAD

**Issue by zwep** [junyanz/pytorch-CycleGAN-and-pix2pix#1001]
Hi,

I have a question about the implementation of the `backward_G` function of the pix2pix model. In the file `models/pix2pix_model.py` I see the following

def backward_G(self):
        """Calculate GAN and L1 loss for the generator"""
        # First, G(A) should fake the discriminator
        fake_AB = torch.cat((self.real_A, self.fake_B), 1)
        pred_fake = self.netD(fake_AB)
        self.loss_G_GAN = self.criterionGAN(pred_fake, True)
        ...

Here we call on `criterionGAN` to make a prediction for a fake input, but we set the target value as if it were a `True` input. Shouldnt that be set to False instead? Of not, why not?

Btw. I understand the concept that the generator needs to fool the discriminator. But thought this was done by still giving the right information to the discriminator, at all times.
Now we are just fooling the discriminator by giving wrong information...

**Comment by junyanz** [comment]
We encourage G to produce images that look realistic (True). If you set False, you are encouraging G to produce images that look fake. See Eqn (13) in the GANs [tutorial](https://arxiv.org/pdf/1701.00160.pdf) for more details.
